addons/account/models/account_tax.py

```diff
@@ -95,13 +95,17 @@ def _default_tax_group(self):
         help="Account used to transition the tax amount for cash basis taxes. It will contain the tax amount as long as the original invoice has not been reconciled ; at reconciliation, this amount cancelled on this account and put on the regular tax account.")
     invoice_repartition_line_ids = fields.One2many(string="Distribution for Invoices", comodel_name="account.tax.repartition.line", inverse_name="invoice_tax_id", copy=True, help="Distribution when the tax is used on an invoice")
     refund_repartition_line_ids = fields.One2many(string="Distribution for Refund Invoices", comodel_name="account.tax.repartition.line", inverse_name="refund_tax_id", copy=True, help="Distribution when the tax is used on a refund")
-    tax_fiscal_country_id = fields.Many2one(string='Fiscal Country', comodel_name='res.country', related='company_id.account_tax_fiscal_country_id', help="Technical field used to restrict the domain of account tags for tax repartition lines created for this tax.")
-    country_code = fields.Char(related='company_id.country_id.code', readonly=True)
+    country_id = fields.Many2one(string="Country", comodel_name='res.country', compute='_compute_country_id', readonly=False, store=True, help="The country for which this tax is applicable.")
 
     _sql_constraints = [
         ('name_company_uniq', 'unique(name, company_id, type_tax_use, tax_scope)', 'Tax names must be unique !'),
     ]
 
+    @api.depends('company_id.account_fiscal_country_id')
+    def _compute_country_id(self):
+        for record in self:
+            record.country_id = record.company_id.account_fiscal_country_id
```

Comment:
Because it needs to be editable without impacting the fiscal country set on the company.
Though, doing it like this means changing the fiscal country of the company will change the country of all the existing taxes ... :s I'll see what I can do, probably just overriding create() is better.

Reply:
Done. Let me know your thoughts about the new version :)